types/coin_test.go

```diff
@@ -1,4 +1,4 @@
-package types
+package types_test
```

Comment:
What is the reason of this change?

Reply:
When you use separate test packages you put yourself in the developers' shoes and get (almost) the guarantee that a client using the feature which broke in tests will also break when it's called (long story short: the whole point of using `_test` packages, which incidentally are very Go-idiomatic).

Reply:
I think it's a matter of a taste. With `_test` you are more focused on integration tests, and do potentially imports which could do cycle dependency. Without `_test` you can access private functions and unit tests them.